### ISSUE TITLE
Add "households" query function

### ISSUE THREAD

**Issue by evansd** [opensafely/cohort-extractor#170]
In the first instance I'm imagining something like this:
```py
...
household_id=patients.household_as_of("2020-02-01", returning="pseudo_id"),
household_size=patients.household_as_of("2020-02-01", returning="household_size"),
...
```

Relevant tables are:
![localhost_8888_notebooks_tpp-schema ipynb](https://user-images.githubusercontent.com/19630/84904002-b5d64680-b0a6-11ea-8ff3-b4521e8e470f.png)

**Comment by wjchulme** [comment]
Can this be closed?

**Comment by evansd** [comment]
Ah yes, thanks!

Closed in #177

**Comment by wjchulme** [comment]
Going to re-open this to query UPRN household too. Can put in a new issue if preferred.

`patients.household_as_of` returns household info from the `Household` and `HouseholdMember` tables (based on GP registration address and tidied up a bit with house sales data). Getting household info from the `UPRN` table for comparison would be useful too. Possibly neater to add to the existing function with a new `method` argument? Taking either `"UPRN"` or `"TPP"` or something. So it would be 

     household_id=patients.household_as_of(
         "2020-02-01", returning="pseudo_id", method="UPRN"
     ),
     
     household_size=patients.household_as_of(
         "2020-02-01", returning="household_size", method="UPRN"
     ),

Also Jonny has added some columns that people may want to query, but this is lower priority
https://ebmdatalab.slack.com/archives/C010SJ89SA3/p1596809707010000?thread_ts=1596178671.407500&cid=C010SJ89SA3

**Comment by wjchulme** [comment]
Just to clarify that the household size will come from `UPRN`, but this table will need to be joined to `MPI` on `Pseudo_uprn` (and `Pseudo_parent_uprn`?) to get the patient IDs. 

There may be repeat patient IDs in `MPI` with different households, valid at different times. The easiest way to deal with this is to use `Latest_Flag` to get the most recent known address. And since this is a one-off extract for the moment, leave the `reference_date` argument to do what it does currently.

If in future we want household size at different times, there are date variables in `'MPI` (`Date_Added` ,`DateFrom`, `Original_Posting_Date`) but it's not yet clear exactly what these mean or how reliable they are.

Best to double check all this with Jonny before going live!

**Comment by sebbacon** [comment]
Closed for #372
